ignite/metrics/rouge.py

```diff
@@ -0,0 +1,151 @@
+from collections import defaultdict
+from typing import Callable, DefaultDict, Dict, List, Optional, Union
+
+import torch
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics import Metric
+
+# These decorators helps with distributed settings
+from ignite.metrics.metric import reinit__is_reduced, sync_all_reduce
+
+
+class Rouge(Metric):
+    r"""Calculates the Rouge Score for two Sequence of Tokens.
+
+
+    .. math::
+        F_\1 = \left( 1 - \alpha \right) * \frac{ \text{precision} * \text{recall} }
+        { \left( \alpha * \text{precision} \right) + \text{recall} }
+
+    where :math:`\alpha` is a float between 0 and 1.
+
+    - ``update`` must receive output of the form ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y}``.
+    - `y_pred` must be a Sequence of Tokens.
+    - `y` must be in the form of List of Sequence of Tokens of Model Texts.
+
+    Args:
+        alpha: alpha value for calculating f1 score
+        n: Rouge score to be calculated using n-grams of tokens
+        output_transform (callable, optional): a callable that is used to transform the
+            :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+
+
+    """
+
+    def __init__(self, alpha: float, n: int, device: Union[str, torch.device] = torch.device("cpu")) -> None:
```

Comment:
add argument to `__init__`
```
output_transform: Callable = lambda x: x
```

Reply:
I will. Also, What is the use of that?

Reply:
A way for users to transform output from `Engine` that could be not suitable for the metric. 

See https://pytorch.org/ignite/master/metrics.html#ignite-metrics